src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -966,7 +966,7 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, Compilation
                                 _ = BindToNaturalType(argument, diagnostics);
                                 break;
                             case BoundUnconvertedSwitchExpression { Type: { } naturalType } switchExpr:
-                                _ = ConvertSwitchExpression(switchExpr, naturalType ?? CreateErrorType(), diagnostics);
+                                _ = ConvertSwitchExpression(switchExpr, naturalType ?? CreateErrorType(), targetTyped: false, diagnostics);
```

Comment:
No, because that would produce duplicate diagnostics for a switch expression that has no natural type.  It isn't "the" problem that it has no natural type so that should not be reported as an additional diagnostic.

---
In reply to: [318804193](https://github.com/dotnet/roslyn/pull/38187#discussion_r318804193) [](ancestors = 318804193)